### DISCUSSION TITLE
input and output text?

### DISCUSSION THREAD

**Discussion by joshuamoorexyz** [tui-cs/Terminal.Gui#2090] [Q&A]
I am trying to simply input data from the user and use that inputted value to output to another textview or label. I cannot seem to find any instruction to do so, can someone help?

**Comment by tznind** [comment]
Hi,

Thanks for checking out the library.  The easiest way to get started is to use the `dotnet new` template for gui.cs:

https://github.com/tznind/gui.cs.template#usage

![image](https://user-images.githubusercontent.com/31306100/196105097-a506686a-d102-49e5-a32d-18ee1df19a3a.png)
_Empty dotnet new template application_

There is a [visual designer tool](https://github.com/gui-cs/TerminalGuiDesigner) that lets you create/edit views.  That is probably the easiest way to create a popup input box:

![createinputbox](https://user-images.githubusercontent.com/31306100/196107522-b2ae5eab-a62c-45fa-9930-af5e49c1a2d5.gif)
_Using TerminalGuiDesigner to build an InputBox_

Assuming you have created the above view then you can expose the result and make Ok close the dialog by updating the `MyInputBox.cs` file as follows:

```csharp
public partial class MyInputBox {

    public string Result => textfield1.Text.ToString();

    public MyInputBox() {
        InitializeComponent();

        button1.Clicked += ()=>Application.RequestStop();
    }
}
```
_InputBox.cs_

To change the template program so that the `[ Click Me]` button fires the input box instead of a message box update `MyView.cs` as follows:

```csharp
button1.Clicked += () =>
{
    var box = new MyInputBox();
    Application.Run(box);

    label1.Text = box.Result;
    this.SetNeedsDisplay();
};
```
_MyView.cs_

The finished program would look like:

![happy](https://user-images.githubusercontent.com/31306100/196109286-dce33145-3420-43b3-b7e2-9a9528784810.gif)
